### PR TITLE
dry up transactor try catch

### PR DESCRIPTION
**Pull request by benschac**
## What does this PR do and why?

creating new PR pointed at main

Same PR, I just needed to create a new one because I accidentally merged the last one into `tiny-invariant` and not main 😅
## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by benschac** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1755** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1755" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/1755?utm_source=stack-comment).

File: src/hooks/v1/transactor/SafeTransferFromTx.ts [hunk 1]
```diff
@@ -12,7 +13,15 @@ export function useSafeTransferFromTx(): TransactorInstance<{
   const { projectId, owner } = useContext(V1ProjectContext)
 
   return ({ newOwnerAddress }, txOpts) => {
-    if (!transactor || !projectId || !contracts?.Projects) {
+    invariant(transactor && contracts?.Projects && projectId && owner)
```
Comment (tomquirk) [inline src/hooks/v1/transactor/SafeTransferFromTx.ts:16]: **question** I might need more coffee, but is this `invariant` call supposed to be within the `try` block? That way, when `invariant` throws, we're catching it?

**Review by tomquirk — COMMENTED** [review]
sorry for the delay here @benschac it slipped my radar!

Just left 1 question

File: src/hooks/v1/transactor/SafeTransferFromTx.ts (hunk 1 above)
Comment (benschac) [reply src/hooks/v1/transactor/SafeTransferFromTx.ts:16]: yesss, good catch.

File: src/utils/transactor.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  try {
+    invariant(transactor && contract, 'missing required')
+    const reqArgs = args.filter(arg => !optionalArgs.includes(arg))
+    const isMissingParameter =
+      args.length - optionalArgs.length !== reqArgs.length
+
+    if (isMissingParameter) {
+      throw new Error('missing parameter')
+    }
+
```
Comment (benschac) [inline src/utils/transactor.ts:34]: I rebased and got a bunch of other commits in this branch.


I pushed this branch to more explicitly show work.


I was trying to get the error logic into a function that we could easily use across the board. 

While something like this works @wraeth-eth made a great point that once we start using juice-sdk a lot of this code is going to go away and we're not going to need it.


In the short term, I'm just going to keep fixing these hanging pending states more explicitly in transactor hook calls.


In the long run, I'm sure we'll figure out a better way of write this logic with juice-sdk.

**Review by benschac — COMMENTED** [review]
the one comment I left is the one worth checking. Sorry about the rebase noise.

File: src/utils/transactor.ts (hunk 2 above)
Comment (benschac) [reply src/utils/transactor.ts:34]: cc: @jbx-protocol/frontend-core

**Pull request closed without merging**